python/cuml/tests/dask/test_dask_logistic_regression.py

```diff
@@ -653,3 +658,349 @@ def test_exception_one_label(fit_intercept, client):
+    datatype = np.float32
+    n_parts = 2
+    max_iter = 5  # cannot set this too large. Observed GPU-specific coefficients when objective converges at 0.
+
+    penalty = regularization[0]
+    C = regularization[1]
+    l1_ratio = regularization[2]
+
+    est_params = {
+        "penalty": penalty,
+        "C": C,
+        "l1_ratio": l1_ratio,
+        "fit_intercept": fit_intercept,
+        "max_iter": max_iter,
+    }
+
+    X = np.array(
+        [
+            [-1.1258, -1.1524, -0.2506, -0.4339, 0.5988],
+            [-1.5551, -0.3414, 1.8530, 0.4681, -0.1577],
+            [1.4437, 0.2660, 1.3894, 1.5863, 0.9463],
+            [-0.8437, 0.9318, 1.2590, 2.0050, 0.0537],
+        ],
+        datatype,
+    )
+
+    X = np.ascontiguousarray(X.T)
+    y = np.array([0.0, 1.0, 2.0, 0.0, 1.0], datatype)
+
+    from sklearn.preprocessing import StandardScaler
+
+    scaler = StandardScaler(with_mean=fit_intercept, with_std=True)
+    scaler.fit(X)
+    scaler.scale_ = np.sqrt(scaler.var_ * len(X) / (len(X) - 1))
+    X_scaled = scaler.transform(X)
+
+    X_df, y_df = _prep_training_data(client, X, y, n_parts)
+    from cuml.dask.linear_model import LogisticRegression as cumlLBFGS_dask
+
+    lr_on = cumlLBFGS_dask(standardization=True, verbose=True, **est_params)
+    lr_on.fit(X_df, y_df)
+
+    lron_coef_origin = lr_on.coef_.to_numpy() * scaler.scale_
+    if fit_intercept is True:
+        lron_intercept_origin = lr_on.intercept_.to_numpy() + np.dot(
+            lr_on.coef_.to_numpy(), scaler.mean_
+        )
+    else:
+        lron_intercept_origin = lr_on.intercept_.to_numpy()
+
+    X_df_scaled, y_df = _prep_training_data(client, X_scaled, y, n_parts)
+    lr_off = cumlLBFGS_dask(standardization=False, **est_params)
+    lr_off.fit(X_df_scaled, y_df)
+
+    assert array_equal(lron_coef_origin, lr_off.coef_.to_numpy())
+    assert array_equal(lron_intercept_origin, lr_off.intercept_.to_numpy())
+
+    from cuml.linear_model import LogisticRegression as SG
+
+    sg = SG(**est_params)
+    sg.fit(X_scaled, y)
+
+    assert array_equal(lron_coef_origin, sg.coef_)
+    assert array_equal(lron_intercept_origin, sg.intercept_)
+
+
+@pytest.mark.mg
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize(
+    "regularization",
+    [
+        ("none", 1.0, None),
+        ("l2", 2.0, None),
+        ("l1", 2.0, None),
+        ("elasticnet", 2.0, 0.2),
+    ],
+)
+def test_standardization_sparse_example(fit_intercept, regularization, client):
+    datatype = np.float32
+    n_parts = 2
+    max_iter = 5  # cannot set this too large. Observed GPU-specific coefficients when objective converges at 0.
+
+    penalty = regularization[0]
+    C = regularization[1]
+    l1_ratio = regularization[2]
+
+    est_params = {
+        "penalty": penalty,
+        "C": C,
+        "l1_ratio": l1_ratio,
+        "fit_intercept": fit_intercept,
+        "max_iter": max_iter,
+    }
+
+    X_origin = np.array(
```

Comment:
This is going to lead to tests which are brittle and hard to maintain. Please generate this data and process the naive version of the expected result (you can do the standardization up front and use singe-gpu logistic). Please do this instead of hardcoding values. We seldomly have to resort to doing this but only because a reasonable way to achieve a naive test doesn't exist.

Reply:
Tried generating a random sparse matrix of any size for classification. Please check!